adapter/pkg/mgw/mgw.go

```diff
@@ -73,7 +81,24 @@ const grpcMaxConcurrentStreams = 1000000
 func runManagementServer(server xdsv3.Server, enforcerServer xdsv3.Server, port uint) {
 	var grpcOptions []grpc.ServerOption
 	grpcOptions = append(grpcOptions, grpc.MaxConcurrentStreams(grpcMaxConcurrentStreams))
-	grpcServer := grpc.NewServer()
+
+	cert, err := tlsutils.GetServerCertificate()
+
```

Comment:
Check the behavior when enforcer and router running but adapter killed in the middle.

Reply:
Enforcer also goes down but the router remains in the existing state.